testproject/Assets/Scripts/Testing/ManualNetworkVariableTest.cs

```diff
@@ -54,37 +55,36 @@ private void ValueChanged(int before, int after)
                     int delta = m_TestVar.LocalTick - m_TestVar.RemoteTick;
                     m_Count++;
 
-                    if (m_Count > k_WaitIterations)
+                    if (!m_Valid)
                     {
-                        if (!m_Valid)
+                        m_Valid = true;
+                        m_MinDelta = delta;
+                        m_MaxDelta = delta;
+                        m_LastRemoteTick = m_TestVar.RemoteTick;
+                    }
+                    else
+                    {
+                        m_MinDelta = Math.Min(delta, m_MinDelta);
+                        m_MaxDelta = Math.Max(delta, m_MaxDelta);
+
+                        // tick should not go backward until wrap around (which should be a long time)
+                        if (m_TestVar.RemoteTick == m_LastRemoteTick)
                         {
-                            m_Valid = true;
-                            m_MinDelta = delta;
-                            m_MaxDelta = delta;
-                            m_LastRemoteTick = m_TestVar.RemoteTick;
+                            m_Problems += "Same remote tick receive twice\n";
                         }
-                        else
+                        else if (m_TestVar.RemoteTick < m_LastRemoteTick)
                         {
-                            m_MinDelta = Math.Min(delta, m_MinDelta);
-                            m_MaxDelta = Math.Max(delta, m_MaxDelta);
-
-                            // tick should not go backward until wrap around (which should be a long time)
-                            if (m_TestVar.RemoteTick == m_LastRemoteTick)
-                            {
-                                m_Problems += "Same remote tick receive twice\n";
-                            }
-                            else if (m_TestVar.RemoteTick < m_LastRemoteTick)
-                            {
-                                m_Problems += "Ticks went backward\n";
-                            }
+                            m_Problems += "Ticks went backward\n";
                         }
                     }
                 }
             }
 
             if (m_Count == k_EndIterations)
             {
-                if (m_Problems == "" && Math.Abs(m_MaxDelta - m_MinDelta) < 3)
+                // Let's be reasonable and allow a 5 tick difference
+                // that could be due to timing difference, lag, queueing
+                if (m_Problems == "" && Math.Abs(m_MaxDelta - m_MinDelta) < 5)
```

Comment:
I would tend to replace the '5' as a constant to show meaning, like k_MaxAllowedTickDifference, and then say why '5', but I'm glad to have this comment for now